RobotArdu/src/main/java/de/fhg/iais/roberta/worker/transfer/Bob3TransferWorker.java

```diff
@@ -2,13 +2,21 @@
 
 import de.fhg.iais.roberta.components.Project;
 import de.fhg.iais.roberta.util.Key;
+import de.fhg.iais.roberta.util.Util;
+import de.fhg.iais.roberta.visitor.codegen.NIBOHexPrefix;
 import de.fhg.iais.roberta.worker.IWorker;
 
 public class Bob3TransferWorker implements IWorker {
 
     @Override
     public void execute(Project project) {
-        // Create agent does not require execution of run call and transferring the binary
+        // Check whether a reset is requested. If yes, get the base64 encoded hex of the factory default program.
+        if ( project.getResult() == Key.FIRMWARE_RESET_SUCCESS ) {
+            String path = project.getRobotFactory().getPluginProperties().getCompilerResourceDir() + project.getCompiledProgramPath() +
+                "." + project.getBinaryFileExtension();
+            project.setCompiledHex(Util.getBase64EncodedHex(path, NIBOHexPrefix.getHexPrefixForRobot(project.getRobot())));
+        }
+        // Check whether the robot is connected with create agent program...
         if ( project.getRobotCommunicator().getState(project.getToken()) == null ) {
             project.setResult(Key.ROBOT_PUSH_RUN);
         } else { // otherwise it uses Connector Program
```

Comment:
As we already have a dedicated workflow for "reset" and a "resetFirmware" worker, why not introducing a "transferFirmware" worker

Reply:
Added `NIBOResetFirmwareWorker` instead.